### PR TITLE
feat(MRS): add a new resource to manage data connection

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDataConnection_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs -v

=== RUN   TestAccDataConnection_basic
=== PAUSE TestAccDataConnection_basic
=== CONT  TestAccDataConnection_basic
--- PASS: TestAccDataConnection_basic (619.57s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/mrs	619.608s

```

### CODE REVIEW

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_data_connection_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	getDataConnectionResp, err := getDataConnectionClient.Request("GET", getDataConnectionPath, &getDataConnectionOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataConnection: %s", err)
+	}
+
+	getDataConnectionRespBody, err := utils.FlattenResponse(getDataConnectionResp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving DataConnection: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_data_connection_test.go]:
```suggestion
		return nil, fmt.Errorf("error retrieving data connection: %s", err)
```

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_data_connection.go [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+	if err != nil {
+		log.Printf("[ERROR] error parsing source_info to json= %#v", resp)
+		return rst
+	}
+
+	rst = []interface{}{
+		map[string]interface{}{
+			"db_instance_id": utils.PathSearch("rds_instance_id", curJson, nil),
+			"db_name":        utils.PathSearch("db_name", curJson, nil),
+			"user_name":      utils.PathSearch("user_name", curJson, nil),
```
Comment (ShiChangkuo) [inline huaweicloud/services/mrs/resource_huaweicloud_mapreduce_data_connection.go]:
```suggestion
			"db_instance_id": curJson["rds_instance_id"],
			"db_name":        curJson["db_name"],
			"user_name":      curJson["user_name"],
```

File: huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_data_connection_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/mrs/resource_huaweicloud_mapreduce_data_connection_test.go]: done

File: huaweicloud/services/mrs/resource_huaweicloud_mapreduce_data_connection.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/mrs/resource_huaweicloud_mapreduce_data_connection.go]: done

**Pull request merged**